File: scala/src/HelloBranch.scala
object HelloBranch {
  def main(args: Array[String]): Unit = {
    var helloWorld = "Hello, world!"

    if (helloWorld == "Good morning") {
      println("morning")
    } else if (helloWorld contains "Hello") {
      println("noon")
    } else {
      println("night")
    }

    helloWorld match {
      case "Hello, world!" =>
        println("nonn")
      case "Good morning, world!" =>
        println("morning")
      case other =>
        println("night")
    }
  }
}

File: rust/src/hello_branch.rs
fn main() {
    let hello_world = "Hello, world!";

    if hello_world == "Good morning" {
        println!("morning");
    } else if hello_world.contains("Hello") {
        println!("noon");
    } else {
        println!("night");
    }

    match hello_world {
        "Hello, world!" => println!("nonn"),
        "Good morning, world!" => println!("morning"),
        _ => println!("night"),
    }
}
